catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -536,46 +554,103 @@ public TransactionResponseType transaction(CswTransactionRequest request) throws
     response.setVersion(CswConstants.VERSION_2_0_2);
 
     int numInserted = 0;
+    final Subject subject = SecurityUtils.getSubject();
```

Comment:
Do we need to verify subject was there? 

Looking at SecurityUtils getSubject it looks like it will attempt to create one if there's not one bound to the thread (and presumably throw if it can't create one), so we should be good, but might be good idea to double check my skimming 😄

Reply:
I think this should be ok. The goal is to pass the current subject into the thread so that the parallel processing executes as if it was done from the original request.